joint_calling/upload_processor.py

```diff
@@ -0,0 +1,81 @@
+""" This function prepares gCVF's uploaded to GCS, based off sample status
+    logged in a database, for further QC and downstream analysis.
+    The upload processor will determine when samples should be added
+    to existing MatrixTables where appropriate and which MatrixTables
+    they should be combined with in this case. Following a successful
+    run all uploaded files will be moved to archival storage."""
+
+from google.cloud import storage
+from google.api_core.exceptions import NotFound
```

Comment:
Can we try to make this agnostic to the storage backend? E.g. imagine we're porting this to Azure one day. Can we design the interfaces in a way that we wouldn't need to change the callers?